src/pass.h

```diff
@@ -421,6 +421,16 @@ class WalkerPass : public Pass, public WalkerType {
     WalkerType::walkFunction(func);
   }
 
+  void runOnExpression(PassRunner* runner,
+                       Module* module,
+                       Function* func,
+                       Expression*& curr) {
+    setPassRunner(runner);
+    WalkerType::setModule(module);
+    WalkerType::setFunction(func);
+    WalkerType::walk(curr);
```

Comment:
Would it be possible to `visit` the expression instead of `walk`ing the expression to make revisiting children more targeted?

Reply:
No, a walk is needed. Only when walking do we set up the infrastructure to allow `replaceCurrent` to replace the node, etc.